src/promptflow/promptflow/executor/flow_executor.py

```diff
@@ -776,6 +784,20 @@ def _apply_default_value_for_input(inputs: Dict[str, FlowInputDefinition], line_
                 updated_inputs[key] = value.default
         return updated_inputs
 
+    @staticmethod
+    def _convert_image_to_bytes(inputs: Dict[str, FlowInputDefinition], line_inputs: Mapping) -> Dict[str, Any]:
+        updated_inputs = dict(line_inputs or {})
+        for key, value in inputs.items():
+            if value.type == ValueType.IMAGE:
+                updated_inputs[key] = Image.from_file(updated_inputs[key])
+        return updated_inputs
+
+    def _save_image_from_output(self, output: dict, output_dir: str):
```

Comment:
Seems like we didn't try to save output image file for each node.
Just to make sure that's one known issue.

Reply:
We will save the intermediate images when serializing output.